pennylane/interfaces/jacobian_products.py

```diff
@@ -0,0 +1,173 @@
+
+    @abc.abstractmethod
+    def compute_vjp(self, tapes: Batch, dy: Tuple[Tuple[TensorLike]]) -> Tuple:
+        """Compute the vjp for a given batch of tapes.
+
+        This method is used by autograd, torch, and tensorflow.
+
+        Args:
+            tapes: the batch of tapes to the the derivatives of
+            dy: the derivatives of the results of an execution
+
+        Returns:
+            TensorLike
+
+        >>> tape0 = qml.tape.QuantumScript([qml.RX(0.1, wires=0)], [qml.expval(qml.PauliZ(0))])
+        >>> tape1 = qml.tape.QuantumScript([qml.RY(0.2, wires=0)], [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliX(0))])
+        >>> batch = (tape0, tape1)
+        >>> dy0 = (0.5, )
+        >>> dy1 = (2.0, 3.0)
+        >>> dys = (dy0, dy1)
+        >>> jp_method.compute_vjp(batch, dys)
+        (array([-0.04991671]), array([2.54286107]))
+        >>> 0.5 * -np.sin(0.1)
+        -0.04991670832341408
+        >>> 2.0 * -np.sin(0.2) + 3.0 * np.cos(0.2)
+        2.5428610719336024
+
+        """
+
+    @abc.abstractmethod
+    def compute_jacobian(self, tapes: Batch) -> Tuple:
+        """Compute the full jacobian for a batch of tapes.
+
+        This method is required for jax-jit.
+
+        Args:
+            tapes: the batch of tapes to take the jacobian of
+
+        >>> tape0 = qml.tape.QuantumScript([qml.RX(0.1, wires=0)], [qml.expval(qml.PauliZ(0))])
+        >>> tape1 = qml.tape.QuantumScript([qml.RY(0.2, wires=0)], [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliX(0))])
+        >>> batch = (tape0, tape1)
+        >>> jp_method.compute_jacobian(batch)
+        (array(-0.09983342), (array(-0.19866933), array(0.98006658)))
+
+        """
+
+
+class TransformDerivatives(JacobianProductCalculator):
+    """Compute vjp, jvps, and jacobians via a gradient transform.
+
+    Args:
+        inner_execute (Callable[[Tuple[QuantumTape]], ResultBatch]): a function that
+            turns the batch of circuits into results.
+        gradient_transform (qml.gradients.gradient_transform): the gradient transform to use.
+        gradient_kwargs (dict): Any keyword arguments for the gradient transform.
+
+    >>> inner_execute = qml.device('default.qubit').execute
+    >>> gradient_transform = qml.gradients.param_shift
+    >>> kwargs = {"broadcast": True}
+    >>> jp_method = TransformDerivatives(inner_execute, gradient_transform, kwargs)
+
+    """
+
+    def __repr__(self):
+        return f"TransformDerivatives({self._inner_execute}, gradient_transform={self._gradient_transform}, gradient_kwargs={self._gradient_kwargs})"
+
+    def __init__(
+        self,
+        inner_execute: Callable,
+        gradient_transform: "qml.gradients.gradient_transform",
+        gradient_kwargs: Optional[dict] = None,
+    ):
+        self._inner_execute = inner_execute
+        self._gradient_transform = gradient_transform
+        self._gradient_kwargs = gradient_kwargs or {}
+
+    def execute_and_compute_jvp(self, tapes: Batch, tangents):
```

Comment:
docstring to be added

Reply:
docstring inherits. Nothing needs to be changed from the `JacobianProductCalculator` docstring.